### PR TITLE
fix: referral api timestamp consistency

### LINKED ISSUE #9784: [Bug]: GraphQL inconsistency in type on referral/discount `endOfProgramTimestamp`
### Problem encountered

https://github.com/vegaprotocol/vega/blob/27081f6f1b5ef209ed284f1f9ad221a3e001fd61/datanode/gateway/graphql/schema.graphql#L4333

https://github.com/vegaprotocol/vega/blob/27081f6f1b5ef209ed284f1f9ad221a3e001fd61/datanode/gateway/graphql/schema.graphql#L4366

https://github.com/vegaprotocol/vega/blob/27081f6f1b5ef209ed284f1f9ad221a3e001fd61/datanode/gateway/graphql/schema.graphql#L5990

https://github.com/vegaprotocol/vega/blob/27081f6f1b5ef209ed284f1f9ad221a3e001fd61/datanode/gateway/graphql/schema.graphql#L6010

https://github.com/vegaprotocol/vega/blob/27081f6f1b5ef209ed284f1f9ad221a3e001fd61/datanode/gateway/graphql/schema.graphql#L6251

### Observed behaviour

The type is inconsistent

### Expected behaviour

Data is treated the same

# Tasks
1. Make them all `Timestamp`

### Steps to reproduce

```bash
1. Look at the schema
```

### Software version

0.73.0-preview15*

### Failing test

_No response_

### Jenkins run

_No response_

### Configuration used

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by ettec**
closes #9784

### CODE REVIEW

File: datanode/gateway/graphql/schema.graphql [hunk 1]
```diff
@@ -6004,8 +6004,8 @@ type CurrentReferralProgram {
   version: Int!
   "Defined tiers in increasing order. First element will give Tier 1, second element will give Tier 2, etc."
   benefitTiers: [BenefitTier!]!
-  "Timestamp as RFC3339Nano, after which when the current epoch ends, the program will end and benefits will be disabled."
-  endOfProgramTimestamp: Timestamp!
+  "Timestamp as RFC3339, after which when the current epoch ends, the programs will end and benefits will be disabled."
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:6007]: does this end all programs? why the change to plural?

File: datanode/gateway/graphql/schema.graphql [hunk 2]
```diff
@@ -4360,8 +4360,8 @@ type StakingTier {
 type UpdateVolumeDiscountProgram {
   "The benefit tiers for the program"
   benefitTiers: [VolumeBenefitTier!]!
-  "The end time of the program"
-  endOfProgramTimestamp: Timestamp!
+  "Timestamp as RFC3339, after which program ends."
+  endOfProgramTimestamp: String!
```
Comment (wwestgarth) [inline datanode/gateway/graphql/schema.graphql:4364]: This is the wrong way around, they should all be of type `TimeStamp`.

**Review by candida-d — APPROVED** [review]

**Review by ValentinTrinque — APPROVED** [review]

File: datanode/gateway/graphql/update_referral_program_resolver.go [hunk 3]
```diff
@@ -28,8 +28,9 @@ func (u updateReferralProgramResolver) BenefitTiers(_ context.Context, obj *vega
 	return obj.Changes.BenefitTiers, nil
 }
 
-func (u updateReferralProgramResolver) EndOfProgramTimestamp(_ context.Context, obj *vega.UpdateReferralProgram) (string, error) {
-	return time.Unix(obj.Changes.EndOfProgramTimestamp, 0).String(), nil
+func (u updateReferralProgramResolver) EndOfProgramTimestamp(_ context.Context, obj *vega.UpdateReferralProgram) (int64, error) {
+	endTime := time.Unix(obj.Changes.EndOfProgramTimestamp, 0)
+	return endTime.UnixNano(), nil
```
Comment (mattrussell36) [inline datanode/gateway/graphql/update_referral_program_resolver.go:33]: Does this still return rfc3339? This is what the Timestamp scalar represents in the schema, and would be preferred.

Comment (mattrussell36) [reply datanode/gateway/graphql/update_referral_program_resolver.go:33]: See https://github.com/vegaprotocol/vega/blob/develop/datanode/gateway/graphql/schema.graphql#L4969

Comment (jeremyletang) [reply datanode/gateway/graphql/update_referral_program_resolver.go:33]: Yes this will be returning an rfc3339.

**Review by jeremyletang — APPROVED** [review]
